modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/Cookie.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+const CONSENT_TYPES = {
+	FUNCTIONAL: 'CONSENT_TYPE_FUNCTIONAL',
+	NECESSARY: 'CONSENT_TYPE_NECESSARY',
+	PERFORMANCE: 'CONSENT_TYPE_PERFORMANCE',
+	PERSONALIZATION: 'CONSENT_TYPE_PERSONALIZATION',
+};
+
+const generateCookie = (name, value, {secure, ...options}) =>
+	`${name}=${value}; ${Object.entries(options)
+		.map(([key, value]) => {
+			return `${key}=${value}`;
+		})
+		.join('; ')} ${secure ? 'secure' : ''}`;
+
+const Cookie = {
+	_checkConsent(type) {
+		return type === CONSENT_TYPES.NECESSARY || this.get(type) === 'true';
+	},
+
+	/**
+	 * Gets the current value for a specific cookie
+	 *
+	 * @param {string} name
+	 * @returns {(string|undefined)} Cookie value as a string, undefined if not present
+	 */
+	get(name) {
```

Comment:
@tinycarol would it make sense to enforce consent here as well?

Reply:
We could do it. I found there's two ways to handle it:
* Simple API: when setting a cookie, set the cookie and also store the consent type for that specific cookie. We would then use that consent type when calling the getter:
  ```js
  set(name, value, type, options) {
     if (checkConsent(type)) {
        document.cookie += `name=consent_${name}; value=${type}`;
        document.cookie += `name=${name}; value=${value}`;
     }
     ...
  }
  
  get(name) {
     const consent = getCookie(`consent_${name}`);
     
     if (consent && hasConsent(consent)) {
        return getCookie(name);
     }
  }
  
  
  ```

   This way we can keep the method signature as it is. In this case it's more complex because cookies can be added from both front and back end, so we would have to duplicate that logic in the back end. That's why I'm not convinced.

* Simple code: store nothing and force the consumers of this API to pass the consent type when calling this method (so refactor to `get(name, consentType)`):
  ```js
  set(name, value, type, options) {
     if (checkConsent(type)) {
        document.cookie += `name=${name}; value=${value}`;
     }
     ...
  }
  
  get(name, consent) {
     if (consent && hasConsent(consent)) {
        return getCookie(name);
     }
     // We could even delete the cookie here if it exists
     return undefined;
  }
  
  
  ```
   I don't love it because there's nothing to prevent users from calling `Cookie.set` and `Cookie.get` with different consent types.
   
WDYT @dsanz?